### PR TITLE
FEATURE: New post editing period for >= tl2 users

### PR DESCRIPTION
**Pull request by rimian**
* Adds a new setting and for tl2 editing posts (30 days same as old value)
* Sets the tl0/tl1 editing period as 1 day

### CODE REVIEW

**Comment by eviltrout** [comment]
Is there a relevant meta discussion for this change? I'm looking for some context here.

**Comment by coding-horror** [comment]
See https://meta.discourse.org/t/spam-bots-tricking-discourse-filter-by-editing/101684/20

**Comment by rimian** [comment]
Sorry, I am not sure where to reference things and where to put the context just yet. I'll clear that up before my next PR :)

File: spec/components/guardian_spec.rb [hunk 1]
```diff
@@ -1389,32 +1389,51 @@
...
       end
 
-      context 'post is older than post_edit_time_limit' do
-        let(:old_post) { build(:post, topic: topic, user: topic.user, created_at: 6.minutes.ago) }
+      describe 'post edit time limits' do
 
-        before do
-          SiteSetting.post_edit_time_limit = 5
-        end
+        shared_examples "edit old posts" do
```
Comment (SamSaffron) [inline spec/components/guardian_spec.rb]: can we avoid shared_examples,  I think it just confuses stuff. makes the specs a bit harder to follow.

File: spec/requests/posts_controller_spec.rb [hunk 2]
```diff
@@ -313,8 +313,28 @@
...
       end
 
-      it 'does not allow to update when edit time limit expired' do
+      it 'does not allow TL0 or TL1 to update when edit time limit expired' do
         SiteSetting.post_edit_time_limit = 5
+        SiteSetting.tl2_post_edit_time_limit = 30
+
+        post = Fabricate(:post, created_at: 10.minutes.ago, user: user)
+
+        user.trust_level = 1
```
Comment (SamSaffron) [inline spec/requests/posts_controller_spec.rb]: Maybe `update_columns` should do the trick it avoids a bunch of callbacks making the spec a tiny bit faster.

**Comment by SamSaffron** [comment]
Overall the change looks good to me! minor changes to the specs and we should be good to merge.

File: spec/requests/posts_controller_spec.rb (hunk 2 above)
Comment (rimian) [reply spec/requests/posts_controller_spec.rb]: Done!

**Pull request merged**